src/main/java/org/candlepin/subscriptions/capacity/admin/UsageContextSubscriptionProvider.java

```diff
@@ -67,47 +66,45 @@ public Optional<Subscription> getSubscription(
       String billingAccountId,
       OffsetDateTime subscriptionDate) {
 
-    // If billingAccountId is multipart, find all matching first part (ex: azureTenantId)
-    var searchByBillingAccountId = billingAccountId;
-    if (StringUtils.isNotBlank(billingAccountId) && billingAccountId.contains(";")) {
-      searchByBillingAccountId = billingAccountId.split(";")[0];
-    }
-
     UsageCalculation.Key usageKey =
         new Key(
             productId,
             ServiceLevel.fromString(sla),
             Usage.fromString(usage),
             billingProvider,
-            searchByBillingAccountId);
+            billingAccountId);
 
     // Set start date one hour in past to pickup recently terminated subscriptions
     var start = subscriptionDate.minusHours(1);
     List<Subscription> subscriptions =
         subscriptionSyncController.findSubscriptions(
             Optional.ofNullable(orgId), usageKey, start, subscriptionDate);
 
-    var bestMatchSubscriptions =
-        filterByClosestMatchingMultipartBillingAccountId(subscriptions, billingAccountId);
+    var matchedSubscription =
+        subscriptions.stream()
+            .filter(
```

Comment:
We don't need to do this filtering because the billing_account_id is part of the query so they should all be equal.